### PR TITLE
feat(m365-tab): add m365 tab template

### PR DESCRIPTION
**Pull request by kuojianlu**
Add m365 tab template: `templates/tab/js/m365`, `templates/tab/ts/m365`. Based on current tab template, but update:
1. @microsoft/teams-js version to 2.0.0-beta.2
2. @microsoft/teamsfx version to 0.6.0-beta.0
3. msteams-react-base-component version to 4.0.0-preview5
5. TabConfig.jsx and TabConfig.tsx to use @microsoft/teams-js v2
6. auth-start.html and auth-end.html to use @microsoft/teams-js v2

The templates will be used in frontend plugin as a new scenario.

### CODE REVIEW

**Comment by kuojianlu** [comment]
E2E TEST: N/A

File: templates/tab/js/m365/src/components/sample/Welcome.jsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    const teamsfx = new TeamsFx();
+    return isInTeams ? await teamsfx.getUserInfo() : undefined;
+  })?.data;
+  const userName = userProfile ? userProfile.displayName : "";
+  return (
+    <div className="welcome page">
+      <div className="narrow page-padding">
+        <Image src="hello.png" />
+        <h1 className="center">Congratulations{userName ? ", " + userName : ""}!</h1>
+        <p className="center">Your app is running in your {friendlyEnvironmentName}</p>
```
Comment (kimizhu) [inline templates/tab/js/m365/src/components/sample/Welcome.jsx:52]: I suggest to add the hub name info here, to demo the functionality of MetaOS SDK.

Comment (kuojianlu) [reply templates/tab/js/m365/src/components/sample/Welcome.jsx:52]: Agree, but let me add it in another PR for easy code review.

**Review by kimizhu — APPROVED** [review]

**Review by hund030 — APPROVED** [review]
npm@8 failed to resolve dependency tree. Feel free to fix it in another PR.
```
npm ERR! While resolving: teamsfx-template-tab@0.1.0
npm ERR! Found: @microsoft/teamsfx@0.6.0-beta.0
npm ERR! node_modules/@microsoft/teamsfx
npm ERR!   @microsoft/teamsfx@"0.6.0-beta.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @microsoft/teamsfx@"^0.6.0" from @microsoft/mgt-teamsfx-provider@2.4.1-next.teamsfx.3511fba
npm ERR! node_modules/@microsoft/mgt-teamsfx-provider
npm ERR!   @microsoft/mgt-teamsfx-provider@"2.4.1-next.teamsfx.3511fba" from the root project
```

**Review by swatDong — APPROVED** [review]

**Comment by kuojianlu** [comment]
> npm@8 failed to resolve dependency tree. Feel free to fix it in another PR.
> 
> ```
> npm ERR! While resolving: teamsfx-template-tab@0.1.0
> npm ERR! Found: @microsoft/teamsfx@0.6.0-beta.0
> npm ERR! node_modules/@microsoft/teamsfx
> npm ERR!   @microsoft/teamsfx@"0.6.0-beta.0" from the root project
> npm ERR!
> npm ERR! Could not resolve dependency:
> npm ERR! peer @microsoft/teamsfx@"^0.6.0" from @microsoft/mgt-teamsfx-provider@2.4.1-next.teamsfx.3511fba
> npm ERR! node_modules/@microsoft/mgt-teamsfx-provider
> npm ERR!   @microsoft/mgt-teamsfx-provider@"2.4.1-next.teamsfx.3511fba" from the root project
> ```

Great thanks, will fix it in later PR.

**Pull request merged**